class A extends RuntimeException {
  int i;

  //@ public normal_behavior
  //@ ensures i == 1;
  A() {
    i = 1;
  }
}

class B extends A {}

class C extends B {}

class Main {
  //@ requires args != null;
  public static void main(String[] args) {
    try {
      try {
        int i = 0;
        throw new A();
      } catch (A exc) {
        //@ assert exc.i == 2;
      }
    } catch (B exc) {
      //@ assert exc.i == 2;
    }
  }
}

